Tests/MacroToolkitTests/MacroToolkitTests.swift

```diff
@@ -678,4 +678,221 @@ final class MacroToolkitTests: XCTestCase {
             macros: testMacros
         )
     }
+    
+    func testNormalizedDecriptionMultipleVoid() {
+        let type: `Type` = "((()))"
+        
+        XCTAssertEqual(type.normalizedDescription, "()")
+    }
+    
+    func testIsVoid() {
+        let type: `Type` = "(())"
+        
+        XCTAssert(type.isVoid)
+    }
+    
+    func testIsOptional() {
+        let type1: `Type` = "Int?"
+        let type2: `Type` = "Optional<Int>"
+        let type3: `Type` = "Array<Int>"
+        
+        XCTAssert(type1.isOptional)
+        XCTAssert(type2.isOptional)
+        XCTAssertFalse(type3.isOptional)
+    }
+    
+    func testNormalizationVoid() {
+        let type: `Type` = "((()))"
+
+        let normalizedType = type.normalized()
+
+        XCTAssertEqual(normalizedType.description, "()")
+    }
+    
+    func testNormalizationMultipleVoidWithOptional() {
+        let type: `Type` = "(((((()))?)))"
+
+        let normalizedType = type.normalized()
+
+        XCTAssertEqual(normalizedType.description, "(Optional<()>)")
```

Comment:
The outer layer of brackets should get removed by normalization since `(Optional<()>)` is equivalent to `Optional<()>`.

Reply:
But to complicate things, the brackets in `((Int) -> (Int)) -> String` can't be removed (a bit of an annoying edge case to deal with). I guess the rule might be that if the brackets are around a function type which itself is the input type of another function type then they can't be omitted?

Reply:
Oh, and if the brackets are around an implicitly unwrapped optional function type they can't be removed either.

Reply:
In b4a470c I updated the implementation of tuple normalization to remove brackets around simple types too. This allows for both of `(Optional<()>)` and `((Int) -> (Int)) -> String` to be normalized correctly. 
In the tests, I added a check against the string representations of types, on top of the hardcoded expected nromalization, to be sure we are aligned with how Swift compiler normalize those types.